lib/galaxy/model/__init__.py

```diff
@@ -1136,6 +1136,12 @@ def to_dict(self, view='collection', system_details=False):
         rval = super().to_dict(view=view)
         rval['tool_id'] = self.tool_id
         rval['history_id'] = self.history_id
+        if self.workflow_invocation_step:
+            rval['created_by_workflow'] = True
+            rval['workflow_invocation_id'] = self.workflow_invocation_step.workflow_invocation.id
+            rval['workflow_id'] = self.workflow_invocation_step.workflow_invocation.workflow_id
+        else:
+            rval['created_by_workflow'] = False
```

Comment:
This is really muddying the waters here, can you please remove this?

Reply:
While my initial reaction is that I don't like this, of course this would be useful, but this is a bit of a "details" view that we don't want to return whenever someone hits the job api endpoint.

I would suggest adding another view here and then serializing the workflow invocation step (`rval['workflow_invocation_step'] = self.workflow_invocation_step.to_dict()`). You can to add the `workflow_invocation_id` in https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/model/__init__.py#L5670-L5671.

The job component can then fetch the additional details from the vuex store if it is appropriate.

Reply:
this is also hopefully been addressed